#### About
> Hypothesis testing

- It is a statistical method used to determine whether a hypothesis about a population is supported by the data. 
- It involves two types of hypothesis - 
1. Null hypothesis - It is the default hypothesis, which assumes that theere is no significant difference between two or more groups.
2. Alternative hypothesis - It is the one we want to test, which assumes that there is a significant difference between two or more groups.

- For example - Let's say we want to test whether the avg. height of men and women is significantly different. The null hypothesis would be that there is no significant diff in the avg height between men and women, while the alternative hypothesis would be that there is a significant difference.

- Steps to conduct a hypothesis test
1. State the null and alternate hypotheis
- Null hypothesis(H0) - The two groups are not significantly different.
- Alternative hypothesis(Ha) - The two groups are significantly different.
2. Choose a significance level(alpha)  to determine the test. Common alpha values are 0.05,0.01 and 0.001.
3. Calculate data from the sample.
4. Calculate the test statistics - The test statistic depends on the type of test being performed(t-test, z-test or chi-squared) and the data being analysed.
5. Determine the p -value - The p-value is the probability of obtaining a test statistic as extreme as the one observed, assuming that the null hypo. is true.
6/ Compare the p-value with chosen significance level - If the p-value is less than or equal to the significance level, we reject the null hypothesis and accept the alternative hypothesis. If the p-value is greater than the significance level, we fail to reject the null hypothesis.


> T-test statistic

- A t-test statistic is a mesaure used in statistical hypothesis testing to determine if there is a significant difference betweent he mean of two groups. It is used when the pop stand dev is not known and the sample size is rel. small (less than 30)

- T-test compares the difference between means of two groups to the variations within the groups. It measures the difference between the sample means divided by the std. error of the diff bet. the means. The resulting value is t-value.

- The t-value tells us how many std. error the diff between the two means is away from zero. The larger the t-value, The greater the evidence against the null hypothesis i.e the means of the two groups are not sig. different. The t-value is compared to a critical value from a t-distribution with degrees of freedom equal to the sample size minus 2 and the p-value is calculated as prob of obt. a t value as extreme or more extreme than the observed value.

- In summary, The t-test statistics is a measure used to det. the significance of the diff bet the means of two groups, taking into account the variation within the groups.

In [1]:
import pandas as pd
import numpy as np

# Create a sample dataframe with two groups - Male and Female
group1 = np.random.normal(10, 2, 50)
group2 = np.random.normal(12, 2, 50)
df = pd.DataFrame({"group1": group1, "group2": group2})

# Print the dataframe
print(df.head())

      group1     group2
0  10.475052  10.816451
1   9.661347  12.822416
2   9.802450  12.648570
3  14.769537  10.839135
4   9.976487  12.766182


In [4]:
from scipy import stats
# Calculate the t-test statistic and p-value
t_stat, p_value = stats.ttest_ind(group1, group2)

# Print the results
print("t-statistic:", t_stat)
print("p-value:", p_value)


t-statistic: -4.743618519181408
p-value: 7.1408968036579575e-06


In this case, the t-statistic is -4.7436. The negative sign indicates that the mean of the first group is lower than the mean of the second group. The absolute value of the t-statistic (i.e., ignoring the sign) indicates the magnitude of the difference between the means.

The p-value is a measure of the probability of obtaining a t-value as extreme or more extreme than the observed value, assuming the null hypothesis is true. In this case, the p-value is 7.14, which is a very small value (less than 0.05). This suggests that it is highly unlikely to observe such an extreme difference between the means of the two groups if there were no true difference between the populations. Therefore, we can reject the null hypothesis that there is no significant difference between the means of the two groups, and conclude that there is strong evidence to suggest that the means are significantly different